### PR TITLE
Add type field to "ScriptEvaluateResult" for "success" or "exception"

### LINKED ISSUE #339: Add type field to "ScriptEvaluateResult" for "success" or "exception"
Found when discussing issue #274.

Right now the `ScriptEvaluateResult` could contain a `result` or an `exception`. The client can only differentiate between those depending on which field is set. There might be cases when no field or both fields are set.

To make it obvious for the client such a `type` field should be added to the `ScriptEvaluateResult` field.

### PR DESCRIPTION
**Pull request by whimboo**
Fixes #339.

This change needs an update of the wpt tests that I can work on if the change is fine.

***
<a href="https://pr-preview.s3.amazonaws.com/whimboo/webdriver-bidi/pull/340.html" title="Last updated on Dec 1, 2022, 3:59 PM UTC (827fa68)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webdriver-bidi/340/a056c3a...whimboo:827fa68.html" title="Last updated on Dec 1, 2022, 3:59 PM UTC (827fa68)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -4162,12 +4162,14 @@ ScriptEvaluateResult = (
 )
 
 ScriptEvaluateResultSuccess = {
+  type: "result",
```
Comment (sadym-chromium) [inline index.bs:4165]: Maybe `success`?

**Review by sadym-chromium — APPROVED** [review]
LGTM with  `type: "success"`

File: index.bs (hunk 1 above)
Comment (whimboo) [reply index.bs]: I fixed that.

**Comment by whimboo** [comment]
@sadym-chromium given that [changing the behavior of the webdriver bidi client for wdspec tests](https://github.com/web-platform-tests/wpt/blob/d994afbbb6f37fc5d0ca722b621e8637cf6749a8/tools/webdriver/webdriver/bidi/modules/script.py#L101-L103) to correctly check for that type field would break all of the evaluation tests for Chrome do you think that you could add that field relatively soon, and we are going to land the test changes when that change hit the beta channel? Or what would be your suggestion?

**Comment by sadym-chromium** [comment]
That's a good idea to land the implementation before the WPT. Let me do that.

**Comment by whimboo** [comment]
@sadym-chromium I've landed the support for the `type` field in Firefox 109. Please let us know which Chrome release will have support for that as well. Thanks!

**Pull request merged**